frontend/pages/payment/details/[transaction_id].tsx

```diff
@@ -0,0 +1,116 @@
+import { GetStaticPaths, GetStaticProps } from 'next'
+import { useTranslation } from 'next-i18next'
+import { serverSideTranslations } from 'next-i18next/serverSideTranslations'
+import { NextSeo } from 'next-seo'
+import Link from 'next/link'
+import { useRouter } from 'next/router'
+import { ReactElement, useEffect, useState } from 'react'
+import Button from '../../../src/components/Button'
+import Main from '../../../src/components/layout/Main'
+import TransactionCancelButton from '../../../src/components/payment/transactions/TransactionCancelButton'
+import TransactionDetails from '../../../src/components/payment/transactions/TransactionDetails'
+import Spinner from '../../../src/components/Spinner'
+import { useUserContext } from '../../../src/context/user-info'
+import { TRANSACTION_INFO_URL } from '../../../src/env'
+import { TransactionDetailed } from '../../../src/types/Payment'
+
+async function getTransaction(txnId: string) {
+  let res: Response
+  try {
+    res = await fetch(TRANSACTION_INFO_URL(txnId), { credentials: 'include' })
+  } catch {
+    throw 'failed-to-load-refresh'
+  }
+
+  if (res.ok) {
+    return await res.json()
+  } else {
+    throw 'failed-to-load-refresh'
+  }
+}
+
+export default function TransactionPage() {
+  const { t } = useTranslation()
+  const router = useRouter()
+  const user = useUserContext()
+
+  const [transaction, setTransaction] = useState<TransactionDetailed>(null)
+  const [error, setError] = useState('')
+
+  // Once router is ready the transaction ID is attainable
+  useEffect(() => {
+    // Must be logged in to see payment confirmation
+    if (!user.info && !user.loading) {
+      router.push('/login')
+    }
+
+    // Router must be ready to access query parameters
+    if (!router.isReady) {
+      return
+    }
+
+    // Once the transaction ID is known, the associated data can be fetched
+    const { transaction_id } = router.query
+
+    // NOTE: Not sure when this will actually happen?
+    if (Array.isArray(transaction_id)) {
+      return
+    }
+
+    getTransaction(transaction_id).then(setTransaction).catch(setError)
+  }, [router, user])
+
+  let content: ReactElement = <Spinner size={200} />
```

Comment:
Is this spinner state even reachable like this?

Reply:
It is will `transaction` is unset yeah, but the conditional branches underneath are admittedly not the most readable (the `else` is nested)

Reply:
Oh right, sorry missed that that else is nested :D